packages/twenty-server/src/engine/core-modules/graphql/hooks/use-graphql-error-handler.hook.ts

```diff
@@ -230,17 +238,51 @@ export const useGraphQLErrorHandlerHook = <
         const headers = context.req.headers;
         const currentMetadataVersion = context.req.workspaceMetadataVersion;
         const requestMetadataVersion = headers[SCHEMA_VERSION_HEADER];
+        const backendAppVersion =
+          options.twentyConfigService.get('APP_VERSION');
+        const appVersionHeaderValue = headers[APP_VERSION_HEADER];
+        const frontEndAppVersion =
+          appVersionHeaderValue && Array.isArray(appVersionHeaderValue)
+            ? appVersionHeaderValue[0]
+            : appVersionHeaderValue;
 
         if (
           requestMetadataVersion &&
           requestMetadataVersion !== `${currentMetadataVersion}`
         ) {
+          options.metricsService.incrementCounter({
+            key: MetricsKeys.SchemaVersionMismatch,
+          });
           throw new GraphQLError(SCHEMA_MISMATCH_ERROR, {
             extensions: {
               userFriendlyMessage: t`Your workspace has been updated with a new data model. Please refresh the page.`,
             },
           });
         }
+
+        if (
+          frontEndAppVersion &&
+          backendAppVersion &&
+          semver.valid(frontEndAppVersion) &&
+          semver.valid(backendAppVersion)
+        ) {
+          if (
+            compareVersionMajorAndMinor(
+              frontEndAppVersion,
+              backendAppVersion,
+            ) !== 'equal'
```

Comment:
I feel like if we were to deploy the server too early before the frontend, there's nothing we could do. So this message should only be shown if backend version is newer than the frontend version ; but if the frontend version is newer then that's probably our fault and there's nothing they can do so we probably don't want to catch it